File: RUBY/DataStructures/queue.rb
class Queue
    def initialize
        @queue = []
    end

    def enqueue(el)
        queue.unshift(el)
        el
    end

    def dequeue
        queue.pop
    end

    def peek
        queue.last
    end

    private

    attr_reader :queue
end

wo = Queue.new
p wo.enqueue("lavish")

File: JavaScript/DataStructures/queue.js
function Queue() {
    this.queue = []
}

Queue.prototype.enqueue = function(el) {
    this.queue.push(el);
    return el;
}

Queue.prototype.dequeue = function() {
    this.queue.shift();
}

Queue.prototype.peek = function() {
    return this.queue[0];
}

const curry = new Queue();
console.log(curry.enqueue("fire"));
curry.dequeue();



